crates/adapters/src/server/mod.rs

```diff
@@ -476,53 +477,158 @@ async fn input_endpoint(
     response
 }
 
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Eq, ToSchema)]
+pub enum EgressMode {
+    /// Continuously monitor the output of the query.
+    ///
+    /// For queries that support snapshots, e.g.,
+    /// [neighborhood](`OutputQuery::Neighborhood`) queries,
+    /// the endpoint outputs the initial snapshot followed
+    /// by a stream of deltas.  For queries that don't support
+    /// snapshots, the endpoint outputs the stream of deltas
+    /// relative to the current output of the query.
+    #[serde(rename = "watch")]
+    Watch,
+    /// Output a single snapshot of query results.
+    ///
+    /// Currently only supported for [quantile](`OutputQuery::Quantiles`)
+    /// and [neighborhood](`OutputQuery::Neighborhood`) queries.
+    #[serde(rename = "snapshot")]
+    Snapshot,
+}
+
+impl Default for EgressMode {
+    /// If `mode` is not specified, default to `Watch`.
+    fn default() -> Self {
+        Self::Watch
+    }
+}
+
+/// URL-encoded arguments to the `/egress` endpoint.
 #[derive(Debug, Deserialize)]
 struct EgressArgs {
-    // #[serde(default = "HttpOutputTransport::default_mode")]
-    // mode: HttpIngressMode,
+    /// Query to execute on the table.
+    #[serde(default)]
+    query: OutputQuery,
+
+    /// Output mode.
+    #[serde(default)]
+    mode: EgressMode,
+
+    /// Data format used to encode the output of the query, e.g., 'csv',
+    /// 'json' etc.
     #[serde(default = "HttpOutputTransport::default_format")]
     format: String,
+
+    /// For [`quantiles`](`OutputQuery::Quantiles`) queries:
+    /// the number of quantiles to output.
+    #[serde(default = "dbsp::operator::sample::default_quantiles")]
```

Comment:
often you don't want all the quantiles, but only the first or the last one.
maybe there should be an optimization for that case.